### ISSUE TITLE
OpenWRT Version Compatibility Question

### ISSUE THREAD

**Issue by phillipsja97** [imp1sh/ansible_managemynetwork#4]
Hello. Love the project and hope to be able to use it in some way. This depends entirely on if you could answer my question though as I'm currently not running 22.01 version or above of OpenWRT. 

The project mentions that the minimum required OpenWRT version is 22.01. I'm currently using a GL.iNet Flint 2 router and it uses it's own modified version of OpenWRT's 21.02 Snapshot. For the most part, it looks like the majority of the roles would still be possible as they do still exist, but I know that doesn't necessarily mean they'd be compatible. For instance, I'm pretty sure any routing wouldn't be able to be configured as I believe 21.02 used `iptables` and 22.01+ is using `nftables`. Things like this I may be able to reverse engineer but more major under the hood incompatibilities is where it'd get to be to much to want to work on to get mine to work. 

Is the recommended version of 22.01 because anything less is incompatible completely and the project wouldn't run? Or is it a situation where some roles would be affected and others would still work? 

I'm not against forking the project and seeing if I can make it work, and if I did I could submit a PR if you'd like, but it'd depend on just how many differences and incompatibilities would exist. Or, if it's just some roles that would be incompatible maybe they are ones that I don't want to use anyways and I just proceed as normal for the time being. If it turned out that it did have major issues with incompatibility then yeah, I'd probably have to leave it be but out of the few Ansible collections I've seen for OpenWRT I really love the verboseness of your project. Hopefully it's possible, but either way, thanks for a great project. :)

**Comment by imp1sh** [comment]
Well that's kind of hard to answer. The biggest issue most probably would be DSA as OpenWrt transitioned to it a while ago. I know there where other issues during the last couple years but I can't remember. I suggest you do not use Gl.Inet's own breed of OpenWrt but use stock OpenWrt. I don't think there's anything you'd loose.
If you don't like that idea for whatever reason I suggest you just give it a try. If things don't work out just do a reset. If you find incompatibilites I'd be glad if you'd tell me.
Basically everytime a new version of OpenWrt arises the collection will be adjusted to be compatible. Sadly I cannot guarantee backwards compatibility. So generally speaking it's always best to stay with the most current version of OpenWrt.

**Comment by phillipsja97** [comment]
Thanks for the reply .

I _think_ I'll give it a shot and see what happens. I have plans to eventually flash the latest OpenWRT to replace GL.iNet's snapshot but I planned to do this at a later date when I had more time. Might have to just push that timeline up. Thanks again, I'll let you know if I give it a try and if I find anything out.
